src/test/java/BaseTest.java

```diff
@@ -1,11 +1,111 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
 
+import java.time.Duration;
+
+
 public class BaseTest {
 
+    public static WebDriver driver = null;
+
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
-}
\ No newline at end of file
+
+    @BeforeMethod
+
+    public static void launchBrowser() {
+        //      Added ChromeOptions argument below to fix websocket error
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+        options.addArguments("--disable-popup-blocking");
+        options.addArguments("--disable-notifications");
+        driver = new ChromeDriver(options);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+    }
+
+    @AfterMethod
+
+    public static void closeBrowser() {
+        driver.quit();
+    }
+
+    public static void navigateToPage() {
+        String url = "https://qa.koel.app/";
+        driver.get(url);
+
+    }
+
+    public static void provideEmail(String email){
+        WebElement emailField = driver.findElement(By.xpath("//input[@type='email']"));
+        emailField.clear();
+        emailField.sendKeys(email);
+    }
+
+    public static void providePassword(String password){
+        WebElement passwordField = driver.findElement(By.xpath("//input[@type='password']"));
+        passwordField.clear();
+        passwordField.sendKeys(password);
+    }
+
+    public static void clickSubmit() throws InterruptedException {
+        WebElement submitButton = driver.findElement(By.xpath("//button[@type='submit']"));
+        submitButton.click();
+        Thread.sleep(2000);
+
+    }
+
+    public void searchSong(String song) throws InterruptedException{
+        WebElement searchField = driver.findElement(By.xpath("//input[@type='search']"));
+        searchField.sendKeys(song);
+        searchField.click();
+        Thread.sleep(2000);
+    }
+    public void clickViewAllButton() throws InterruptedException {
+        WebElement viewAllWebelement = driver.findElement(By.cssSelector("div.results section.songs h1 button"));
```

Comment:
By the way, in viewAllWebelement selector you can using attributes such as `[data-test='view-all-songs-btn']` to selector more concise

Reply:
Hi Coach, 

I have made the necessary changes and formatted the Code as well. Can you check if it looks fine?

Thanks,
Bhakti